docker/docker_image_dest.go

```diff
@@ -314,7 +305,12 @@ func (d *dockerImageDestination) TryReusingBlob(ctx context.Context, info types.
 		}
 
 		// Whatever happens here, don't abort the entire operation.  It's likely we just don't have permissions, and if it is a critical network error, we will find out soon enough anyway.
-		d.c.extraScope = &authScope{
+
+		// Checking candidateRepo, and mounting from it, requires an
+		// expanded token scope.  We still want to reuse the ping
+		// information and other aspects of the client, so rather than
+		// make a fresh copy, there is this a bit ugly extraScope hack.
```

Comment:
The “extraScope hack” originally referred to the stateful `.extraScope` member and the way it was _not_ passed through `makeRequest`*; this PR passes the data in a more accurate straightforward way, so I think the reference to the “ugly hack” can be dropped.

Unless you want to leave it around to refer to the fair amount of difficult to read parameters to `makeRequest`— but that’s mostly the fault of Go not having named parameters :)

Reply:
:rofl: dropping sounds fine. Thanks!

Reply:
Done :-)